db/blip_handler.go

```diff
@@ -1082,6 +1084,9 @@ func (bh *blipHandler) processRev(rq *blip.Message, stats *processRevStats) (err
 	if deltaSrcRevID, isDelta := revMessage.DeltaSrc(); isDelta && !revMessage.Deleted() {
 		if !bh.sgCanUseDeltas {
 			return base.HTTPErrorf(http.StatusBadRequest, "Deltas are disabled for this peer")
+		} else if !bh.useHLV() {
```

Comment:
For an incoming delta that has a revTreeID as deltaSrc, we should be able to accept if when deltaSrc is the current revTreeID.  That case shouldn't require any temporary revision bodies.

Reply:
Good point I think as long as the delta source rev is in the rev cache its fine, I think it makes sense for me to remove this upon reflection. If we receive a rev delta source, look in rev cache for it and fail to find it (the fail to load as no backup by rev id, the function will error anyway on line 1110.